config/runtime.exs

```diff
@@ -225,14 +261,26 @@ if config_env() != :test do
   config :grpc, port: System.get_env("LOGFLARE_GRPC_PORT", "50051") |> String.to_integer()
 end
 
+socket_options_for_url = fn
+  url when is_binary(url) ->
+    case URI.parse(url) do
+      %URI{host: host} -> socket_options_for_host.(host)
+      _ -> raise "Failed to parse URL: #{inspect(url)}"
```

Comment:
URI.parse always succeeds but there is no guarantee the returned URL will have a host, so perhaps you want this?

```suggestion
      %URI{host: host} when is_binary(host) -> socket_options_for_host.(host)
      _ -> raise "Failed to parse URL with host: #{inspect(url)}"
```